include/RAJA/scan.hxx

```diff
@@ -218,40 +162,17 @@ void inclusive_scan(Iter begin, Iter end, IterOut out)
 template <typename ExecPolicy,
           typename Iter,
           typename IterOut,
-          typename BinaryFunction,
-          typename T>
+          typename T = typename std::iterator_traits<Iter>::value_type,
+          typename BinaryFunction = ::RAJA::operators::plus<T>>
```

Comment:
`T` and `BinaryFunction` can be deduced or defaulted based on usage for all scan operations